packages/runtime/src/module/index.ts

```diff
@@ -0,0 +1,132 @@
+import { getFMId, safeToString } from '../utils';
+import { assert } from '../utils/logger';
+import { getRemoteEntry } from '../utils/load';
+import { FederationHost } from '../core';
+import { Global } from '../global';
+import {
+  RemoteEntryExports,
+  Options,
+  Remote,
+  ShareInfos,
+  RemoteInfo,
+} from '../type';
+import { composeKeyWithSeparator } from '@module-federation/sdk';
+
+export type ModuleOptions = ConstructorParameters<typeof Module>[0];
+
+type HostInfo = Remote;
+
+class Module {
+  hostInfo: HostInfo;
+  remoteInfo: RemoteInfo;
+  inited = false;
+  shared: ShareInfos = {};
+  remoteEntryExports?: RemoteEntryExports;
+  lib: RemoteEntryExports | undefined = undefined;
+  loaderHook: FederationHost['loaderHook'];
+  // loading: Record<string, undefined | Promise<RemoteEntryExports | void>> = {};
+
+  constructor({
+    hostInfo,
+    remoteInfo,
+    shared,
+    loaderHook,
+  }: {
+    hostInfo: HostInfo;
+    remoteInfo: RemoteInfo;
+    shared: ShareInfos;
+    plugins: Options['plugins'];
+    loaderHook: FederationHost['loaderHook'];
+  }) {
+    this.hostInfo = hostInfo;
+    this.remoteInfo = remoteInfo;
+    this.shared = shared;
+    this.loaderHook = loaderHook;
+  }
+
+  async getEntry(): Promise<RemoteEntryExports> {
+    if (this.remoteEntryExports) {
+      return this.remoteEntryExports;
+    }
+
+    // Get remoteEntry.js
+    const remoteEntryExports = await getRemoteEntry({
+      remoteInfo: this.remoteInfo,
+      remoteEntryExports: this.remoteEntryExports,
+      createScriptHook: (url: string) => {
+        const res = this.loaderHook.lifecycle.createScript.emit({ url });
```

Comment:
I think we can use this for node as well, createScript would just need the node target stuff from DFS. This is great